pkg/platform/cicd/service.go

```diff
@@ -0,0 +1,116 @@
+package cicd
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/dolittle-entropy/platform-api/pkg/platform"
+	"github.com/dolittle-entropy/platform-api/pkg/utils"
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+type service struct {
+	logContext      logrus.FieldLogger
+	k8sDolittleRepo platform.K8sRepo
+}
+
+func NewService(logContext logrus.FieldLogger, k8sDolittleRepo platform.K8sRepo) *service {
+	s := &service{
+		logContext:      logContext,
+		k8sDolittleRepo: k8sDolittleRepo,
+	}
+	return s
+}
+
+func (s *service) GetServiceAccountCredentials(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	applicationID := vars["applicationID"]
+	userID := r.Header.Get("User-ID")
+	tenantID := r.Header.Get("Tenant-ID")
+
+	allowed := s.k8sDolittleRepo.CanModifyApplicationWithResponse(w, tenantID, applicationID, userID)
+	if !allowed {
+		return
+	}
+
+	logContext := s.logContext.WithFields(logrus.Fields{
+		"method":      "GetServiceAccountCredentials",
+		"credentials": "serviceAccount",
+		"userID":      userID,
+	})
+	logContext.Info("requested credentials")
+
+	secretName := "azure-devops"
+	secret, err := s.k8sDolittleRepo.GetSecret(applicationID, secretName)
+	if err != nil {
+		if !k8serrors.IsNotFound(err) {
+			logContext.WithFields(logrus.Fields{
+				"error":      err,
+				"secretName": secretName,
+			}).Error("issue talking to cluster")
+
+			utils.RespondWithError(w, http.StatusInternalServerError, "Something went wrong")
+			return
+		}
+
+		logContext.WithFields(logrus.Fields{
+			"error":      err,
+			"secretName": secretName,
+			"hint":       "azure-devops-missing",
+		}).Error("application")
```

Comment:
This is something that I've been wondering for a while now and it might just be me not being familiar with structured logging or logging in go overall.

But I'm wondering why is the error message `application` and why is there a `hint` property (that is describing the error like the error message should?) instead of just writing out a descriptive error message like:
```suggestion
		logContext.WithFields(logrus.Fields{
			"error":      err,
			"secretName": secretName,
		}).Error("couldn't find the azure-devops secret")
```

I see this elsewhere in the codebase too (really short and non-descriptive error messages) and I'm always a bit confused about what are our logging practices in platform-api.

Should we start using a `hint` property instead of the `msg` property of the error message to understand it? To me the error message should already be the "hint" of the error (alongside with all of the other nice properties we put into the structured logs).

Reply:
- I have removed hint.
- Moved the logContext into repo.GetSecrets
- Using a generic "secret not found"